브랜치 'develop/web'의 Server 배포 설정 완료

youngxpepp opened the pull request:
# 브랜치 'develop/web'의 Server 배포 설정 완료

## 📎 해당 이슈

이슈 링크 ( #131 )

## 🛠 구현 내용 

- 네이버 클라우드 배포
- PM2 무중단 배포

## 🙋‍ 리뷰어 참고 사항 
- https://engineering.linecorp.com/ko/blog/pm2-nodejs/
- 118.67.133.30:5000
- develop/web에 merge 되면 테스트를 거쳐서 네이버 클라우드에 자동 배포되도록 설정했습니다.

Park-SooYeon commented on server/src/main.js:
```diff
@@ -2,13 +2,18 @@ import { Application } from './Application';
     await app.listen(process.env.PORT || 5000);
-    process.send('ready');
+
+    if (process?.send ?? false) {
```
어떤 경우에 어플리케이션이 child process로 실행되지 않는 건가요?👀

Park-SooYeon reviewed (dismissed):
고생하셨습니다!:+1:
깃헙 액션 설정파일은 좀 더 공부하고나서 한번 더 천천히 읽어봐야할거같네요..😵

youngxpepp committed:
[FEAT] 무중단 서비스를 위한 PM2 설정

- 어플리케이션이 child process로 실행되지 않는다면 process.send가 undefined인 경우가 존재함
- process.send의 유무 검사 후에 실행할 수 있도록 함
- PM2 graceful start/shutdown 설정
- ecosystem.config.js 추가
- SIGINT 신호를 받았을 때 이벤트 추가

youngxpepp committed:
[FEAT] 브랜치 'develop/web'의 server 배포 설정

- SCP로 빌드된 결과물을 네이버 클라우드에 전송
- SSH 명령어로 PM2 실행

dh00n reviewed (approved):
수고하셨습니당 👍
